File: uva_team/3n+1.cpp
#include <iostream>  
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n){
  vector<int> vec;
  while (n != 1)
  {
    if (n % 2 == 1) n = (3 * n) + 1;
    else n = n / 2;
    vec.push_back(n);
  }
  vec.push_back(1);
  return vec.size();
}

int main(){
  int x, y;

  while( cin >> x >> y) {
    int ans = 0;

    int min = x < y ? x : y;
    int max = x > y ? x : y;

    for( int i = min; i <= max; i++){
      int count = solution(i);
      if( count > ans) ans = count;
    }
    cout << x << " " << y << " " << ans;
    cout << endl;
  }

}

File: uva_team/3n+1.py

def solution(n):
  count = 1
  while n > 1:
    n = ((3*n) + 1) if (n%2 == 1) else (n//2)
    count += 1
  return count

while True:
  try:
    data = input()
    print(data)
    data = [int(x) for x in input().split(" ")]
    ans = 0
    for i in range(data[0], data[1]+1):
      ans = max(solution(i),ans)
    print('{} {} {}'.format(data[0],data[1],ans))
  except EOFError:
    break
  except TypeError:
    print("failed")
    break
  except:
    break
